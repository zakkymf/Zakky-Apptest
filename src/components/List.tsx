import React from 'react';
import {
  Text,
  StyleSheet,
  TouchableOpacity,
  StyleProp,
  ImageStyle,
} from 'react-native';
import {Colors, Fonts} from '../theme';
import Avatar from './Avatar';

interface ListProps {
  name: string;
  photo: string;
  avatarStyle?: StyleProp<ImageStyle>;
  onPress: () => void;
}

const List: React.FC<ListProps> = ({name, photo, onPress, avatarStyle}) => {
  return (
    <TouchableOpacity style={styles.list} onPress={onPress}>
      <Avatar source={photo} avatarStyle={avatarStyle} />
      <Text style={styles.name}>{name}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  list: {
    paddingVertical: 5,
    alignItems: 'center',
    flexDirection: 'row',
  },
  name: {
    fontSize: 14,
    marginLeft: 15,
    color: Colors.black,
    fontFamily: Fonts.type.monserrat,
  },
});

export default List;
