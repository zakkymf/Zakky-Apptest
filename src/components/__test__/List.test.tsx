import React from 'react';
import {render} from '@testing-library/react-native';
import List from '../List';

describe('@components/List', () => {
  it('render list component', () => {
    const component = () =>
      render(
        <List
          name="Zakky Muhammad Fajar"
          photo="https://www.helpforenglish.cz/files/wiki.jpeg"
          onPress={() => jest.fn()}
        />,
      );

    const {toJSON} = component();
    expect(toJSON()).toMatchSnapshot();
  });
});
