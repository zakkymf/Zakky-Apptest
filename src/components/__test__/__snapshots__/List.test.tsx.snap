// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@components/List render list component 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "paddingVertical": 10,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://www.helpforenglish.cz/files/wiki.jpeg",
      }
    }
    style={
      Object {
        "borderRadius": 20,
        "height": 40,
        "width": 40,
      }
    }
  />
  <Text
    style={
      Object {
        "color": "#000000",
        "fontFamily": "Montserrat-Regular",
        "fontSize": 14,
        "marginLeft": 15,
      }
    }
  >
    Zakky Muhammad Fajar
  </Text>
</View>
`;
